enhancements/machine-api/control-plane-machine-set.md

```diff
@@ -0,0 +1,1032 @@
+    // ControlPlaneMachineSet. It corresponds to the ControlPlaneMachineSets's generation,
+    // which is updated on mutation by the API Server.
+    // +optional
+    ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+    // Replicas is the number of Control Plane Machines created by the
+    // ControlPlaneMachineSet controller.
+    // Note that during update operations this value may differ from the
+    // desired replica count.
+    Replicas int32 `json:"replicas"`
+
+    // ReadyReplicas is the number of Control Plane Machines created by the
+    // ControlPlaneMachineSet controller which are ready.
+    ReadyReplicas int32 `json:"readyReplicas,omitempty"`
+
+    // UpdatedReplicas is the number of non-terminated Control Plane Machines
+    // created by the ControlPlaneMachineSet controller that have the desired
+    // provider spec.
+    UpdatedReplicas int32 `json:"updatedReplicas,omitempty"`
+
+    // UnavailableReplicas is the number of Control Plane Machines that are
+    // still required before the ControlPlaneMachineSet reaches the desired
+    // available capacity. When this value is non-zero, the number of
+    // ReadyReplicas is less than the desired Replicas.
+    UnavailableReplicas int32 `json:"unavailableReplicas,omitempty"`
+
+    // Conditions represents the observations of the ControlPlaneMachineSet's current state.
+    // Known .status.conditions.type are: (TODO)
+    // TODO: Identify different condition types/reasons that will be needed.
+    // +patchMergeKey=type
+    // +patchStrategy=merge
+    // +listType=map
+    // +listMapKey=type
+    // +optional
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+// FailureDomain represents the different configurations required to spread Machines
+// across failure domains on different platforms.
+// +union
+type FailureDomain struct {
+  // Platform identifies the platform for which the FailureDomain represents
+  // +kubebuilder:validation:Enum:="aws";"azure";"gcp";"openstack"
+  // +unionDiscriminator
+  // +optional
+  Platform configv1.PlatformType `json:platform,omitempty`
+
+  // AWS configures failure domain information for the AWS platform
+  // +optional
+  AWS *AWSFailureDomain `json:aws,omitempty`
+
+  // Azure configures failure domain information for the Azure platform
+  // +optional
+  Azure *AzureFailureDomain `json:azure,omitempty`
+
+  // GCP configures failure domain information for the GCP platform
+  // +optional
+  GCP *GCPFailureDomain `json:gcp,omitempty`
+
+  // OpenStack configures failure domain information for the OpenStack platform
+  // +optional
+  OpenStack *OpenStackFailureDomain `json:openstack,omitempty`
+}
+
+// AWSFailureDomain configures failure domain information for the AWS platform
+type AWSFailureDomain struct {
+  // Subnet is a reference to the subnet to use for this instance
+  // +optional
+	Subnet AWSResourceReference `json:"subnet,omitempty"`
```

Comment:
I think there might be mixed indentation here. And in a couple of places below too.

Reply:
Good catch